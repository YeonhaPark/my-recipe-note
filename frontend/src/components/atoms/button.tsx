import { ReactNode } from 'react';
import {
  Button as MButton,
  ButtonTypeMap,
  ButtonProps,
} from '@material-ui/core';
import { withStyles } from '@material-ui/styles';

export const IButton = <
  D extends React.ElementType = ButtonTypeMap['defaultComponent'],
  P = {},
>(
  props: ButtonProps<D, P>,
) => {
  return <MButton {...props} />;
};
interface Props {
  onClick?: () => void;
  children: string | ReactNode;
  variant?: 'text' | 'contained' | 'outlined';
  fullWidth?: boolean;
  style?: any;
  color?:
    | 'inherit'
    | 'primary'
    | 'secondary'
    | 'success'
    | 'error'
    | 'info'
    | 'warning'
    | 'basic'
    | undefined;
}

const CustomButton = withStyles({
  root: {
    boxShadow: 'none',
  },
})(IButton);

export default function Button({
  onClick,
  children,
  color,
  variant,
  fullWidth,
  style,
}: Props): JSX.Element {
  return (
    <CustomButton
      onClick={onClick}
      color={color}
      style={style}
      variant={variant}
      fullWidth={fullWidth}
    >
      {children}
    </CustomButton>
  );
}

Button.defaultProps = {
  style: {},
  fullWidth: false,
  color: 'inherit',
  variant: 'text',
};
