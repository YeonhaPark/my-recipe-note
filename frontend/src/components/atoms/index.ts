import Button from './button';
import Chip, { ChipType, ChipColor } from './chip';
import Checkbox from './checkbox';

export { Button, Chip, Checkbox };
export type { ChipType, ChipColor };
