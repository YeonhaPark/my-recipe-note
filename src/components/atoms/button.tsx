import { ReactNode } from 'react';
import { Button as MButton } from '@material-ui/core';
import { withStyles } from '@material-ui/styles';

interface Props {
  children: string | ReactNode;
  variant: 'text' | 'contained' | 'outlined';
  fullWidth: boolean;
  style?: any;
}

const CustomButton = withStyles({
  root: {
    boxShadow: 'none',
    marginBottom: '1.5rem',
  },
})(MButton);

export default function Button({
  children,
  variant,
  fullWidth = false,
  style,
}: Props): JSX.Element {
  return (
    <CustomButton style={style} variant={variant} fullWidth={fullWidth}>
      {children}
    </CustomButton>
  );
}

Button.defaultProps = {
  style: {},
};
