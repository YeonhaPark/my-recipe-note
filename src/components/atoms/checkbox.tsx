import { withStyles } from '@material-ui/styles';
import { Checkbox as MCheckbox } from '@material-ui/core';

const CustomCheckbox = withStyles({
  root: {
    paddingTop: 0,
    paddingBottom: 0,
  },
})(MCheckbox);

interface Props {
  id: number;
  checked: boolean;
  onCheckboxChange: (
    type: string,
    id: number,
    e: React.ChangeEvent<HTMLInputElement>,
  ) => void;
  name: string;
  color: 'primary' | 'secondary';
}
export default function Checkbox({
  id,
  checked,
  onCheckboxChange,
  name,
  color,
}: Props) {
  return (
    <CustomCheckbox
      checked={checked}
      onChange={(e) => onCheckboxChange('checkbox', id, e)}
      name={name}
      color={color}
    />
  );
}
