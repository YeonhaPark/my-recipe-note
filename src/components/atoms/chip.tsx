import Chip from '@material-ui/core/Chip';
import Stack from '@material-ui/core/Stack';

export default function ColorChips(): JSX.Element {
  return (
    <Stack spacing={1} alignItems="center">
      <Stack direction="row" spacing={1}>
        <Chip onClick={() => {}} label="korean" color="warning" />
        <Chip label="japanese" color="secondary" />
      </Stack>
      <Stack direction="row" spacing={1}>
        <Chip label="mexican" color="warning" variant="outlined" />
        <Chip label="fastfood" color="secondary" variant="outlined" />
      </Stack>
    </Stack>
  );
}
