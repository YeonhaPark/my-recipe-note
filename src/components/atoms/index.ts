import Button from './button';
import Chip from './chip';

export { Button, Chip };
