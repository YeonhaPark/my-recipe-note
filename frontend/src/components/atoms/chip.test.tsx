import { mount } from 'enzyme';
import { Chip, ChipColor } from '../atoms';
import renderer from 'react-test-renderer';

const chipProps = {
  label: 'korean',
  color: 'info' as ChipColor,
};

const renderChip = () => mount(<Chip {...chipProps} />);

describe('Chip', () => {
  it('renders Chip', () => {
    renderChip();
  });

  it('renders label passed from parent', () => {
    const chip = renderChip();
    chip.setProps({ label: 'chinese' });
    expect(chip.text()).toEqual('chinese');
  });

  it('matches snapshot', () => {
    const tree = renderer.create(<Chip {...chipProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
