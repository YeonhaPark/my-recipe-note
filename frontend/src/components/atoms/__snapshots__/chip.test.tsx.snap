// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip matches snapshot 1`] = `
<span
  className="css-1loe067-CustomChip"
>
  <div
    className="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorInfo MuiChip-filledInfo css-1ifqgi5-MuiChip-root"
    onKeyDown={[Function]}
    onKeyUp={[Function]}
  >
    <span
      className="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
    >
      korean
    </span>
  </div>
</span>
`;
